aws-xray-recorder-sdk-benchmark/tst/main/java/com/amazonaws/xray/entities/EntityBenchmark.java

```diff
@@ -51,75 +76,79 @@
         @Setup(Level.Trial)
         public void setupOnce() throws SocketException {
             recorder = AWSXRayRecorderBuilder.defaultRecorder();
+            segments = new ArrayList<>();
+            annotationMap = new HashMap<>();
+            metadataMap = new HashMap<>();
             theException = new Exception("Test Exception");
-        }
+            cause = new Cause();
 
-        @Setup(Level.Invocation)
-        public void setup() {
-            parentSegment = new SegmentImpl(recorder, SEGMENT_NAME);
+            for (int i = 0; i < N_OPERATIONS; i++) {
+                segments.add(new SegmentImpl(recorder, SEGMENT_NAME));
+            }
         }
 
         @TearDown(Level.Invocation)
         public void doTearDown() {
```

Comment:
I think just running the setup on every iteration would be simpler than this, and avoids needing to add public API `setCause`

Reply:
Running setup every iteration or invocation?

Reply:
Sorry meant invocation (both start with i and end with n!)

Reply:
Agreed!